test/libyul/yulOptimizerTests/fullSuite/medium.yul

```diff
@@ -23,7 +23,12 @@
 //
 // {
 //     {
-//         sstore(0, add(mload(0x40), 128))
-//         sstore(1, 0x20)
+//         let p := mload(0x40)
+//         mstore(0x40, add(p, 96))
+//         let _1 := add(p, 128)
+//         mstore(_1, 2)
+//         mstore(0x40, 0x20)
+//         sstore(0, _1)
+//         sstore(1, mload(0x40))
```

Comment:
This one is caused by `LoadResolver` (`L`) not being able to resolve `mload(0x40)` when the unused `mstore(0x80, 0x40)` stands between it and `mstore(0x40, 0x20)`.

In the default sequence the `xa[r]cL` segment can deal with it because it runs after the structural simplification segment, which includes `StructuralSimplifier` (`t`) and `UnusedStoreEliminator` (`S`). The new sequence instead encounters this:


```yul
            if 1
            {
                if 1 { mstore(0x40, 0x20) }
                for { } true { }
                {
                    if 1 { break }
                    mstore(0x80, 0x40)
                }
            }
            sstore(0, _1)
            sstore(1, mload(0x40))
```

Replacing `xa[r]cL` with `txa[r]cSL` fixes the problem and `mload(0x40)` gets resolved into `0x20`.
Note that `t` here is ineffective if we run it on SSA form while the opposite is true for `S`.

Reply:
Oddly, while the change helps with this test and also minimally decreases gas usage in several other tests, there are also a few where it makes things a little worse.

For example for `functionCall/external_call_to_nonexisting.sol` bytecode size seems to grow enough to increase the code deposit cost by almost 2%:
```diff
-// gas irOptimized: 89315
-// gas irOptimized code: 169800
+// gas irOptimized: 89578
+// gas irOptimized code: 173000
```

So my proposed fix has mixed effects. I'll have to see the effect on external tests to see if it's actually worth it.

Reply:
### Comparison with unmodified `the-good-parts-mk2` on `cancun`
#### Tweak
```diff
-       "xa[r]cL"                      // Turn into SSA again and simplify
+       "txa[r]cSL"                    // Turn into SSA again and simplify
```

#### External tests (`ir-optimize-evm+yul`)
[`seqbench-sequence-the-good-parts-mk2-tweaks-t-and-c-before-load-resolver`](https://app.circleci.com/pipelines/github/ethereum/solidity/33672/workflows/7c9f885c-f2e9-430a-8c13-037c6d522599) vs [`seqbench-sequence-the-good-parts-mk2-cancun`](https://app.circleci.com/pipelines/github/ethereum/solidity/33655/workflows/f901ee55-094e-44b1-860c-44cab6b675bb)

|          project |  bytecode_size | deployment_gas |     method_gas |
|:----------------:|---------------:|---------------:|---------------:|
|        elementfi | **`-0.09% ✅`** |                |                |
|              ens | **`+0.25% ❌`** | **`+1.42% ❌`** | **`-0.02% ✅`** |
|            euler | **`+0.05% ❌`** |                |                |
|              gp2 | **`+0.08% ❌`** |                |                |
|  perpetual-pools |           `!A` |           `!A` |           `!A` |
|    pool-together | **`+0.05% ❌`** |                |                |
|          uniswap | **`+0.03% ❌`** |                |                |
| yield_liquidator | **`+0.03% ❌`** | **`+0.01% ❌`** |          `-0%` |
|         zeppelin | **`+0.06% ❌`** |  **`+0.1% ❌`** | **`+0.22% ❌`** |

For `perpetual-pools` compilation fails with `YulException: Variable _16 is 1 too deep in the stack.` (memoryguard was present).

#### Gas diff stats
| File name                                                       | IR optimized   | Legacy optimized   | Legacy   |
|-----------------------------------------------------------------|----------------|--------------------|----------|
| `functionCall/external_call_to_nonexisting.sol`              | 1%             |                    |          |
| `inlineAssembly/transient_storage_selfdestruct.sol`          | 1%             |                    |          |
| `functionCall/external_call_to_nonexisting_debugstrings.sol` | 1%             |                    |          |
| `functionCall/gas_and_value_brace_syntax.sol`                | 1%             |                    |          |
| `functionCall/gas_and_value_basic.sol`                       | 1%             |                    |          |